provisioning/virtual-machines/ansible/roles/gns3/tasks/main.yml

```diff
@@ -75,3 +75,108 @@
     owner: "{{ ansible_user }}"
     group: "{{ ansible_user }}"
     mode: '0644'
+
+### The following tasks are due to the previous manual setup of GNS3 environment
+
+## 1.  Download the Cisco IOS Image
+
+# 1.1 Actual download 
+- name: Download the binary file
+  command: |
+          curl -o /home/{{ ansible_user }}/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.bin
+          https://owncloud.ipv6.polito.it:8080/index.php/s/OobvABlroRZcEWJ/download
+
+# 1.2 Decompress the file from .bin to .image
+- name: Unzip the file
+  shell: > 
+        unzip -p /home/crownlabs/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.bin > 
+        /home/crownlabs/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.image
+  ignore_errors: yes
+
+# 1.3 Remove the compressed .bin file
+- name: Remove the archive file
+  file:
+    path: /home/crownlabs/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.bin
+    state: absent
+
+## 2.  Create a new template for the Cisco 2691 router
+
+# GNS3 server is usually launched though UI. A terminal launch is needed.
+
+# 2.1 Create a gns3.service file which is going to be handled via systemd
+- name: Copy the gns3 service file
+  copy:
+    src: "{{ role_path }}/files/gns3.service"
+    dest: /etc/systemd/system/gns3.service
+    mode: '0755'
+
+# 2.2 Enable and start the new gns3 service
+- name: Enable and start the gns3 service
+  systemd:
+    name: gns3
+    state: started
+    enabled: yes
+
+# 2.3 The GNS3 server needs some seconds to be ready
+- name: Introduce a sleep delay of 10 seconds
+  pause:
+    seconds: 5
```

Comment:
5 or 10 seconds?

Reply:
As you prefer, in any case though, I see nothing wrong with waiting 5 more seconds, if it means to be safer and avoid wasting maybe 15 minutes in rebuilding from scratch :)

Reply:
The process should ideally take just a couple of seconds. Striking a balance between security and responsiveness, a 5-second timeframe would be appropriate.